files/hipchat.rb

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env ruby
+
+require "#{File.dirname(__FILE__)}/base"
+#require_relative 'base'
+
+class Hipchat < BaseHandler
+  def handle
+    timeout_and_retry do
+      case @event['check']['status'].to_i
+      when 1,2
+        trigger_incident
+      when 0
+        resolve_incident
+      end
+    end
+  end
+
+  def api_key
+    handler_settings['api_key'] || false
+  end
+
+  def trigger_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour, :notify =>  true }
+    )
+  end
+
+  def resolve_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour }
+    )
+  end
+
+  def event_time
+    Time.at(@event['check']['issued']).utc.strftime "%Y-%m-%d %H:%M:%S UTC"
+  end
+
+  def hipchat_message
+"
+<b>#{event_time} - #{@event['check']['name']} on #{@event['client']['name']} (#{@event['client']['address']}) - #{human_check_status}</b>
+<br /><br />
+&nbsp;&nbsp;#{@event['check']['notification'] || @event['check']['output']}
+"
+  end
+
+  def hipchat_message_colour
+    case @event['check']['status']
+    when 0
+      'green'
+    when 1
+      'yellow'
+    when 2
+      'red'
+    else
+      'grey'
+    end
+  end
+
+  def default_room
+    handler_settings['hipchat_room']
+  end
+
+  def hipchat_room
+    team_data('hipchat_room') || default_room || nil
```

Comment:
This is super! My only regret here is how I setup the `irc_channels` thing originally.

One of the design goals of these handlers is that we could build tools that "didn't have to remember that ops uses opsgenie instead of pagerduty". 

Here we are making a new key for hipchat, and if anyone wanted to build a tool programmatically alert teams, that tool would either have to spam both irc/hipchat/slack/etc to make sure to get it all, or it would have to "know" which teams did what.

I don't like this, and I shouldn't have baked in the word "irc" into `irc_channels` because it was bad precedent.

I understand that hipchat has "rooms", irc and slack have channels. What do you think about responding to just "channels" and then I'll do the work of migrating the irc handler to use the same key?